modules/varscan/1.0/varscan.smk

```diff
@@ -0,0 +1,194 @@
+    _varscan_symlink_maf,
+    _varscan_output_vcf,
+    _varscan_output_maf,
+    _varscan_dispatch,
+    _varscan_all,
+
+
+##### RULES #####
+
+# Symlinks the input files into the module results directory (under '00-inputs/')
+rule _varscan_input_bam:
+    input:
+        bam = CFG["inputs"]["sample_bam"]
+    output:
+        bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam"
+    run:
+        op.relative_symlink(input.bam, output.bam)
+        op.relative_symlink(input.bam + ".bai", output.bam + ".bai")
+
+
+rule _varscan_bam2mpu:
+    input:
+        bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam",
+        fasta = reference_files("genomes/{genome_build}/genome_fasta/genome.fa")
+    output:
+        mpu = temp(CFG["dirs"]["mpileup"] + "{seq_type}--{genome_build}/{sample_id}.mpileup")
+    log:
+        stderr = CFG["logs"]["mpileup"] + "{seq_type}--{genome_build}/{sample_id}.bam2mpu.stderr.log"
+    params:
+        opts = CFG["options"]["mpileup"]
+    conda:
+        CFG["conda_envs"]["samtools"]
+    threads:
+        CFG["threads"]["mpileup"]
+    resources:
+        mem_mb = CFG["mem_mb"]["mpileup"]
+    shell:
+        op.as_one_line("""
+        samtools mpileup {params.opts}
+        -f {input.fasta} {input.bam}
+        > {output.mpu}
+        2> {log.stderr}
+        """)
+
+
+rule _varscan_somatic:
+    input:
+        tumour_mpu = CFG["dirs"]["mpileup"] + "{seq_type}--{genome_build}/{tumour_id}.mpileup",
+        normal_mpu = CFG["dirs"]["mpileup"] + "{seq_type}--{genome_build}/{normal_id}.mpileup"
+    output:
+        snp = CFG["dirs"]["varscan"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/snp.vcf",
+        indel = CFG["dirs"]["varscan"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/indel.vcf"
+    wildcard_constraints:
+        pair_status = "matched|unmatched"
+    log:
+        stdout = CFG["logs"]["varscan"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/varscan_somatic.stdout.log",
+        stderr = CFG["logs"]["varscan"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/varscan_somatic.stderr.log"
+    params:
+        opts = op.switch_on_wildcard("seq_type", CFG["options"]["somatic"])
+    conda:
+        CFG["conda_envs"]["varscan"]
+    threads:
+        CFG["threads"]["somatic"]
+    resources:
+        mem_mb = CFG["mem_mb"]["somatic"]
+    shell:
+        op.as_one_line("""
+        varscan somatic 
+        {input.normal_mpu} {input.tumour_mpu} 
+        --output-snp {output.snp} --output-indel {output.indel}
+        {params.opts}
+        > {log.stdout} 2> {log.stderr}
+        """)
+
+
+rule _varscan_unpaired:
+    input:
+        tumour_mpu = CFG["dirs"]["mpileup"] + "{seq_type}--{genome_build}/{tumour_id}.mpileup",
+    output:
+        vcf = CFG["dirs"]["varscan"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{vcf_name}.vcf"
```

Comment:
Would it be a problem downstream if these were gzipped as they were created?

Reply:
I'm not sure I can try that as well

Reply:
I'm wondering if we should keep the output consistent (.vcf or .vcf.gz) between variant caller modules, since it would simplify the vcf2maf module. I think vcf2maf only takes non-gsipped vcf files, but correct me if I'm wrong.

Reply:
I haven't tried it on gzipped directly but the module should be able to decompress them temporarily.

Reply:
Come to think of it, they should probably be "bgzip2" compressed (which is what Strelka2 does by default)